trin-core/src/portalnet/types.rs

```diff
@@ -555,10 +559,327 @@ impl FromStr for HexData {
     }
 }
 
+// ===========================================
+// CONTENT KEY FUNCTIONALITY (will be moved to separate file)
+// ====================================
```

Comment:
I'd rather see this move happen in this pr. I can't think of a great reason to delay the move for a later pr (but i'm all ears). But yes, in regards to your other comment, it makes sense to keep them inside `trin-core` if `trin-core` depends on these types.